File: src/org/cerion/symcalc/expression/function/statistics/StandardDeviation.java
package org.cerion.symcalc.expression.function.statistics;

import org.cerion.symcalc.expression.Expr;
import org.cerion.symcalc.expression.function.Function;
import org.cerion.symcalc.expression.function.FunctionExpr;

public class StandardDeviation extends FunctionExpr {

    public StandardDeviation(Expr... e) {
        super(Function.STANDARD_DEVIATION,e);
    }

    @Override
    protected Expr evaluate() {
        return null;
    }
}

File: symcalc/src/main/kotlin/function/statistics/StandardDeviation.kt
package org.cerion.symcalc.function.statistics

import org.cerion.symcalc.expression.Expr
import org.cerion.symcalc.function.FunctionExpr
import org.cerion.symcalc.function.arithmetic.Sqrt

class StandardDeviation(vararg e: Expr) : FunctionExpr(*e) {

    // FEAT Has more advanced usages, matrix/dist parameters

    override fun evaluate(): Expr {
        val list = get(0).asList()
        val variance = Variance(list)

        return Sqrt(variance).eval()
    }

    override fun validate() {
        validateParameterCount(1)
        validateParameterType(0, ExprType.LIST)
    }
}
